clients/typescript/src/types.ts

```diff
@@ -1,19 +1,89 @@
+/**
+ * @typedef {Record<string, unknown>} StanVariableInputs
+ * A type holding named inputs to a Stan model,
+ * e.g. the data or initial values.
+ */
 export type StanVariableInputs = Record<string, unknown>;
 
+/**
+ * @typedef {Function} PrintCallback
+ * A callback for printing output from the Stan model.
+ * @param {string} s The string to print.
+ * @returns {void}
+ */
 export type PrintCallback = (s: string) => void;
 
+/**
+ * The metric used for the HMC sampler.
+ * @enum {number}
+ * @readonly
+ * @property {0} UNIT - Unit metric.
+ * @property {1} DENSE - Dense metric.
+ * @property {2} DIAGONAL - Diagonal metric.
+ */
 export enum HMCMetric {
   UNIT = 0,
   DENSE = 1,
   DIAGONAL = 2,
 }
 
+/**
+ * @typedef {Object} StanDraws
+ * A type holding the result of a Stan sampling run.
+ * @property {string[]} paramNames The names of the parameters in
+ * the model.
+ * @property {number[][]} draws A 2D array of draws from the posterior.
+ * The first dimension is the number of samples, and the second dimension
+ * is the number of parameters.
+ * @property {number[][] | number[][][]} metric The metric used for the
+ * HMC sampler. If the metric is not saved, this field is not present.
+ */
 export type StanDraws = {
   paramNames: string[];
   draws: number[][];
   metric?: number[][] | number[][][];
 };
 
+/**
+ * @typedef {Object} SamplerParams
+ * Parameters for the HMC sampler.
+ * @property {string | StanVariableInputs} [data=""] The data for the model.
```

Comment:
These defaults ( and others that are just quotes) are getting rendered in the Github version as an HTML escape (`&quot;&quot;`) instead of the actual values.

Reply:
Yep, see https://github.com/godaddy/dmd/issues/6

Like I said, it's just the least broken one I found (do _not_ tempt me to write my own entire dmd template...)